sw/airborne/modules/flight_plan_guided/flight_plan_guided.c

```diff
@@ -0,0 +1,352 @@
+ * @param[in] heading, rotate to the wanted heading in [rad] (clockwise is positive)
+ * @param[return] Boolean for flight plan to keep running (true) or to just run once (false)
+ */
+bool RotateToHeading(float heading)
+{
+  if (autopilot.mode != AP_MODE_GUIDED) { return true; }
+
+  guidance_h_set_guided_heading(heading);
+  return false;
+}
+
+/**
+ * SpeedRotateToHeading: Will slowly turn to a given heading with a fixed speed
+ * @param[in] rotation_rate, speed to slowly rotate the heading(clockwise is positive)
+ * @param[in] heading, here the MAV will stop spinning [rad] (clockwise is positive)
+ *  @param[return] Boolean for flight plan to keep running (true) or to just run once (false)
+ */
+bool SpeedRotateToHeading(float rotation_rate, float heading)
+{
+  guidance_h_set_guided_heading_rate(rotation_rate);
+
+  if (fabs(wrapToPi_guided(heading) - stateGetNedToBodyEulers_f()->psi) < 0.1) {
+    guidance_h_set_guided_heading(heading);
+    return false;
+  }
+  return true;
+}
+
+/********************************************//**
+ *  Blocking conditions
+ ***********************************************/
+
+/**
+ * WaitUntilAltitude: Blocks while waiting until a certain altitude is reached
+ * @param[in] altitude in [m] (positive z-axis is up)
+ * @param[return] Boolean for flight plan to keep running (true) or to just run once (false)
+ */
+bool WaitUntilAltitude(float altitude)
+{
+  if (autopilot.mode != AP_MODE_GUIDED) { return true; }
+  if (stateGetPositionEnu_f()->z < altitude) { return true; }
+
+  return false;
+}
+
+
+/**
+ * WaitUntilAltitude: Blocks while waiting until a certain altitude is reached
+ * @param[in] altitude in [m] (positive z-axis is up)
+ * @param[return] Boolean for flight plan to keep running (true) or to just run once (false)
+ */
+bool WaitUntilGround(void)
+{
+  if (autopilot.mode != AP_MODE_GUIDED) { return true; }
+  if (stateGetPositionEnu_f()->z > 0.1) { return true; }
+
+  return false;
+}
+
+/**
+ * WaitUntilSpeedOrAltitude: Blocks while waiting until a certain altitude or speed is reached
+ * @param[in] fail_altitude, wait for altitude in [m] (positive z-axis is up)
+ * @param[in] speed, speed in [m/s] (positive z-axis is up)
+ * @param[return] Boolean for flight plan to keep running (true) or to just run once (false)
+ */
+bool WaitUntilSpeedOrAltitude(float speed, float altitude)
+{
+  if (autopilot.mode != AP_MODE_GUIDED) { return true; }
+
+  if (stateGetPositionEnu_f()->z < altitude && stateGetSpeedEnu_f()->z < speed) { return true; }
+
+  return false;
+}
+
+/**
+ * WaitforHeading: Blocks untill wanted heading is achieved
+ * @param[in] heading, waits untill MAV reached certain heading [rad] (clockwise is positive)
+ *  @param[return] Boolean for flight plan to keep running (true) or to just run once (false)
+ *
+ * TODO: Wrap the angle around -pi and pi
+ */
+bool WaitforHeading(float heading)
+{
+  if (fabs(wrapToPi_guided(heading) - stateGetNedToBodyEulers_f()->psi) < 0.1) {
```

Comment:
This test could be implement in the `NavCourseCloseTo` that is currently a void macro in `firmwares/rotorcraft/navigation.h`